sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -139,6 +139,21 @@ def as_provider_list(name, lst):
   return [as_provider(name, x) for x in lst]
 
 
+def parse_callable_kwargs(spec: Optional[Mapping[str, Any]]):
+  def _parse_callable_kwargs_rec(args, top_level=False):
+    parsed_args = {}
+    for arg, val in args.items():
+      parsed_args[arg] = val
```

Comment:
Nit: Prefer to write once, rather than assign and then re-assign (e.g. put this in an else).